src/peer_channels.rs

```diff
@@ -0,0 +1,597 @@
+
+        Ok(connect_future)
+    }
+
+    /// Cache a peers [NodeId] for one topic.
+    pub(crate) async fn iroh_add_peer_for_topic(
+        &self,
+        msg_id: MsgId,
+        topic: TopicId,
+        peer: NodeId,
+        relay_server: Option<&str>,
+    ) -> Result<()> {
+        self.sql
+            .execute(
+                "INSERT OR REPLACE INTO iroh_gossip_peers (msg_id, public_key, topic, relay_server) VALUES (?, ?, ?, ?)",
+                (msg_id, peer.as_bytes(), topic.as_bytes(), relay_server),
+            )
+            .await?;
+        Ok(())
+    }
+
+    /// Get a list of [NodeAddr]s for one webxdc.
+    async fn get_iroh_gossip_peers(&self, msg_id: MsgId) -> Result<Vec<NodeAddr>> {
+        self.sql
+            .query_map(
+                "SELECT public_key, relay_server FROM iroh_gossip_peers WHERE msg_id = ? AND public_key != ?",
+                (msg_id, self.get_iroh_node_addr().await?.node_id.as_bytes()),
+                |row| {
+                    let key:  Vec<u8> = row.get(0)?;
+                    let server: Option<String> = row.get(1)?;
+                    Ok((key, server))
+                },
+                |g| {
+                    g.map(|data| {
+                        let (key, server) = data?;
+                        let server = server.map(|data| Ok::<_, url::ParseError>(RelayUrl::from(Url::parse(&data)?))).transpose()?;
+                        let id = NodeId::from_bytes(&key.try_into()
+                        .map_err(|_| anyhow!("Can't convert sql data to [u8; 32]"))?)?;
+                        Ok::<_, anyhow::Error>(NodeAddr::from_parts(
+                            id, server, vec![]
+                        ))
+                    })
+                    .collect::<std::result::Result<Vec<_>, _>>()
+                    .map_err(Into::into)
+                },
+            )
+            .await
+    }
+
+    /// Get the iroh secret key from the database or generate a new one and persist it.
+    pub(crate) async fn get_or_generate_iroh_keypair(&self) -> Result<SecretKey> {
+        match self.get_config_parsed(Config::IrohSecretKey).await? {
+            Some(key) => Ok(key),
+            None => {
+                let key = SecretKey::generate();
+                self.set_config_internal(Config::IrohSecretKey, Some(&key.to_string()))
+                    .await?;
+                Ok(key)
+            }
+        }
+    }
+
+    /// Get the iroh [NodeAddr] without direct IP addresses.
+    pub(crate) async fn get_iroh_node_addr(&self) -> Result<NodeAddr> {
+        let endpoint = self.iroh_endpoint.read().await;
+        let relay = endpoint
+            .as_ref()
+            .context("iroh endpoint not initialized")?
+            .my_relay();
+        Ok(NodeAddr::from_parts(
+            endpoint
+                .as_ref()
+                .context("iroh endpoint not initialized")?
+                .node_id(),
+            relay,
+            vec![],
```

Comment:
Could just use `endpoint.my_addr()` and then delete the direct addresses, if you don't want them